test/unit/rules/lint-no-arguments-for-html-elements-test.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const ERROR_MESSAGE = require('../../../lib/rules/lint-no-arguments-for-html-elements')
+  .ERROR_MESSAGE;
+const ERROR_MESSAGE_2 = require('../../../lib/rules/lint-no-arguments-for-html-elements')
+  .ERROR_MESSAGE_2;
+const ERROR_MESSAGE_3 = require('../../../lib/rules/lint-no-arguments-for-html-elements')
+  .ERROR_MESSAGE_3;
+
+generateRuleTests({
+  name: 'no-arguments-for-html-elements',
+
+  config: true,
+
+  good: [
+    '<Input @name=1 />',
+    '{{#let (component \'foo\') as |bar|}} <bar @name="1" as |n|><n/></bar> {{/let}}',
+    '<@externalComponent />',
+    `<MyComponent>
+    <:slot @name="Header"></:slot>
+  </MyComponent>`,
+    '<@foo.bar @name="2" />',
+    '<@foo @name="2" />',
+    '<foo.some.name @name="1"/>',
+  ],
+
+  bad: [
+    {
+      template: '<div as |a|></div>',
+      result: {
+        message: ERROR_MESSAGE_3.replace('%', 'a'),
+        moduleId: 'layout.hbs',
+        source: '<div as |a|></div>',
+        line: 1,
+        column: 0,
+      },
+    },
+    {
+      template: '<div name=@value></div>',
```

Comment:
same as above, this should probably be extracted to a dedicated rule, because

1. it would be useful for component invocations too
2. there might be cases where you actually want to pass a `@`-prefixed string as the attribute and in that case it should be possible to disable the rule

Reply:
any suggestions how this roule could be named? (require-mustache-wrap-variables) ?

 for prefixed string it should be `name="@" or name = '@' ` and we can check it using `sourceForNode` method + indexOf

Reply:
maybe `no-at-prefixed-string-values` or something like that

I'm not sure I understand the second question 🤔

Reply:
2.) solved using this approach -> https://github.com/ember-template-lint/ember-template-lint/pull/927/files#diff-407e0d43d58a271ba8e91f1876ea034fR37